File: algorithms/javascript/9-palindrome-number/solution-1.js
/**
 * Personally I think this one might be nonesense for JS Dev. Because of numbers of string methods to use. Look like cheat, but just a choice.
 * @param {number} x
 * @return {boolean}
 */
 var isPalindrome = function(x) {
    return x.toString().split('').reverse().join('') === x.toString();
};

File: algorithms/9-palindrome-number/python/solution-1.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0 or (x % 10 == 0 and x != 0):
            return False

        reverse = 0
        temp = x

        while temp != 0:
            reverse = reverse * 10 + temp % 10
            temp //= 10

        return reverse == x